packages/protocol/contracts/L1/TaikoData.sol

```diff
@@ -66,9 +66,9 @@ library TaikoData {
         bytes signature;
         uint32 l1StateBlockNumber;
         uint64 timestamp;
-        uint256 blobTxListOffset;
-        uint256 blobTxListLength;
-        uint256 blobIndex;
+        uint64 blobTxListOffset;
+        uint64 blobTxListLength;
+        uint8 blobIndex;
```

Comment:
The choice to use `uint8` may iincrease gas payment, the same to `uint64`.

Reply:
FYI, modifying the code this way is OK for a fresh new testnet, but it will break the current mainnet upgradability, therefore, the code shall never be merged into main.

Reply:
> FYI, modifying the code this way is OK for a fresh new testnet, but it will break the current mainnet upgradability, therefore, the code shall never be merged into main as-is, new BlockParams struct must be defined, please see #17734 to learn how to make breaking changes.

Yep, until something is "locked in" we will just do it like this, to avoid wasting time over engineering.

Reply:
Great, later lets figure out together when/how to do a upgrade, then cherrypick your changes into a new fork-PR.